src/ui/components/components.rs

```diff
@@ -292,13 +292,14 @@ impl Chat {
         // Remove old messages
         self.clear_messages();
         // TODO: Add new messages
-        // let (tx, rx) = oneshot::channel();
-        // runtime().spawn(async move
-        // {
-        //     let messages = ApiEndpoints::GetMessages(channel_id, None, 50).call(todo!()).await.unwrap();
-        //     tx.send(messages).unwrap();
-        // });
-        // let message = rx.blocking_recv().unwrap();
+        let (tx, rx) = oneshot::channel();
+        runtime().spawn(async move
+        {
+            let messages = ApiEndpoints::GetMessages(channel_id, None, 50).call(None).await.unwrap();
+            println!("{:?}", messages);
+            tx.send(messages).unwrap();
+        });
+        let message = rx.blocking_recv().unwrap();
```

Comment:
This should be in open_chat and not in add_channel function. Due to the fact that it is already in open_chat you can just remove it out of here and that should be fine.

Reply:
add_channel doesnt have this function. This is switch_chat